### PR TITLE
Availability elapsed

### PR DESCRIPTION
**Pull request by treykris**
Description
-----------
- Calculate the total amount of hours in the volunteer's availability between the last time they modified their availability and the new time stamp

Developer self-review checklist
-------------------------------
- [x] Potentially confusing code has been explained with comments
- [x] No warnings or errors have been introduced; all known error cases have been handled
- [ ] Any appropriate documentation (within the code, README.md, docs, etc) has been updated
- [x] All edge cases have been addressed

### CODE REVIEW

File: tests/utils/get-frequency-of-days.test.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+test('Start date and end date are the same day', t => {
+  const startDate = new Date('2020-02-06')
+  const endDate = new Date('2020-02-06')
+  const expected = [0, 0, 0, 0, 1, 0, 0]
+  const result = getFrequencyOfDays(startDate, endDate)
+  t.deepEqual(expected, result)
+})
+
+test('Start Date: first of leap year - End Date: first of next year', t => {
```
Comment (treykris) [inline tests/utils/get-frequency-of-days.test.js]: @jkeat,

Some of these tests in this file are failing on Circle CI, but all are passing locally. If you get a chance to run them can you please verify that they pass locally

File: utils/get-frequency-of-days.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  const dayFrequencyList = Array(7).fill(amountOfWeeks)
+  const startDayOfWeek = startDate.day()
+  const endDayOfWeek = endDate.day()
+
+  if (startDayOfWeek <= endDayOfWeek) {
+    // leftover days are to the right of the startDayOfWeek
+    dayFrequencyList.fill(amountOfWeeks + 1, startDayOfWeek, endDayOfWeek + 1)
+  } else {
+    // leftover days are to the left of endDayOfWeek and
+    // leftover days are to the right of startDayOfWeek
```
Comment (jkeat) [inline utils/get-frequency-of-days.js]: Hmm, is one of these comments a duplicate or is it saying there are leftover days to the left and to the right?

**Comment by jkeat** [comment]
@treykris When you get a chance, could you look at the merge conflict?

File: utils/get-frequency-of-days.js (hunk 2 above)
Comment (treykris) [reply utils/get-frequency-of-days.js]: @jkeat,

That both leftover days are to the left and right instead of the range between them

**Comment by treykris** [comment]
@jkeat ,

- Updated to resolve merge conflicts

File: controllers/CalendarCtrl.js [hunk 3]
```diff
@@ -28,8 +28,12 @@ module.exports = {
       return callback(new Error('Availability object missing required keys'))
     }
 
+    const newModifiedDate = new Date().toISOString()
+    user.elapsedAvailability = user.calculateElapsedAvailability(
```
Comment (jkeat) [inline controllers/CalendarCtrl.js]: Should this be `user.elapsedAvailability += user.calculateElapsedAvailability( ... )` ? I think this would overwrite existing values, but we want to keep a running tally.

File: models/User.js [hunk 4]
```diff
@@ -306,7 +312,8 @@ var userSchema = new mongoose.Schema(
       }
     },
     availabilityLastModifiedAt: { type: Date, default: Date.now },
-    lastActivityAt: { type: Date, default: Date.now }
+    lastActivityAt: { type: Date, default: Date.now },
+    elapsedAvailability: Number
```
Comment (jkeat) [inline models/User.js]: Might be good to give this a default of 0

File: controllers/CalendarCtrl.js (hunk 3 above)
Comment (treykris) [reply controllers/CalendarCtrl.js]: @jkeat,

Ah, good catch!
